### PR TITLE
fix(core): stabilize constraint settle and layout shape invariants

### PR DESCRIPTION
**Pull request by RtlZeroMemory**
## Summary
Stabilize Rezi’s constraint/layout engine to remove first-frame/resize jump behavior and eliminate shape/cache drift edge cases in nested and overlay-heavy trees.

## What Changed
- include `constraintGraph.requiredRuntimeInstanceIds` in constraint input signature/key invalidation
- add bounded in-frame settle passes for parent-dependent constraint chains (`max 12`)
- traverse `layer.content` and `modal.content/actions` for constraint + scroll override application
- strengthen runtime/layout shape mismatch detection with stable identity (`id`/`key`) checks and cold-relayout fallback
- fix stack downstream-reservation edge cases in both legacy and advanced flex planning modes
- preserve original flow+absolute child order in `stack` and `box` output trees
- preserve hidden subtree shape (`display:false`) with zero-size layout trees
- align runtime commit semantics for single-child wrappers (`field`, `resizablePanel`)
- include `wrap` in stack layout stability signature
- add regression tests for all above engine behaviors

## Validation
- `npm run -s build`
- `node scripts/run-tests.mjs --filter "widgetRenderer.constraints|layout.flex-distribution|layout.edgecases|layout.stability-signature"`
  - pass: `161`, fail: `0`
- `node scripts/run-tests.mjs`
  - pass: `4911`, fail: `0`, suites: `568`

## PTY / Frame Audit Evidence
- interactive PTY pass (`examples/regression-dashboard/src/main.ts`) with key path: `j`, `t`, `f`, `q`
- `node scripts/frame-audit-report.mjs /tmp/rezi-engine-pr-reg-audit.ndjson --latest-pid`
  - `backend_submitted=4`
  - `worker_payload=4`
  - `worker_accepted=4`
  - `worker_completed=4`
  - `hash_mismatch_backend_vs_worker=0`

## Summary by CodeRabbit

* **Bug Fixes**
  * More reliable multi-pass constraint settlement, deterministic relayout on shape/identity mismatches, and preserved subtree shapes for hidden/display:false containers; fixed child ordering and sizing when mixing absolute and flow children; modal/action ordering and overlay visibility correctness.

* **New Features**
  * Improved flex behavior (flexShrink/flexBasis) and inclusion of wrap in layout stability signatures.

* **Tests**
  * Expanded coverage for constraint settlement, nested/deep layouts, resizing, edge cases, and layout stability.

* **Documentation**
  * Updated changelog with unreleased fixes and test notes.

### CODE REVIEW

File: packages/core/src/app/widgetRenderer.ts [hunk 1]
```diff
@@ -886,6 +913,14 @@ function hasRuntimeLayoutShapeMismatch(root: RuntimeInstance, layoutRoot: Layout
     if (!runtimeNode || !layoutNode) continue;
 
     if (runtimeNode.vnode.kind !== layoutNode.vnode.kind) return true;
+    const runtimeIdentity = readLayoutShapeIdentity(runtimeNode.vnode);
+    const layoutIdentity = readLayoutShapeIdentity(layoutNode.vnode);
+    if (
+      (runtimeIdentity !== null || layoutIdentity !== null) &&
+      runtimeIdentity !== layoutIdentity
+    ) {
```
[the remark this answers is not in the document]
Comment (RtlZeroMemory) [reply packages/core/src/app/widgetRenderer.ts:921]: Implemented. `layoutOverlays` now computes modal action positions from right-to-left but appends resulting action layout nodes in declared order, keeping runtime/layout child identity alignment. Added regression test: `modal layout children preserve declared action order` in `layout.edgecases.test.ts`.

File: packages/core/src/layout/engine/layoutEngine.ts [hunk 2]
```diff
@@ -265,6 +265,59 @@ function getSyntheticThemedColumn(vnode: ThemedVNode): VNode {
...
+      for (let i = 0; i < props.actions.length; i++) {
+        const action = props.actions[i];
+        if (isVNode(action)) children.push(action);
+      }
+    }
+    return children;
+  }
+
+  return [];
+}
```
[the remark this answers is not in the document]
Comment (RtlZeroMemory) [reply packages/core/src/layout/engine/layoutEngine.ts:303]: Implemented. `hiddenLayoutChildrenForVNode` now mirrors runtime single-child wrapper semantics for `field` and `resizablePanel` before generic direct-child extraction. Added regression coverage in `layout.edgecases.test.ts` (`display:false mirrors single-child wrapper semantics for hidden subtrees`).

File: packages/core/src/app/widgetRenderer.ts [hunk 3]
```diff
@@ -2432,17 +2467,16 @@ export class WidgetRenderer<S> {
     return null;
   }
 
-  private applyScrollOverridesToVNode(vnode: VNode): VNode {
+  private applyScrollOverridesToVNode(
+    vnode: VNode,
+    overrides: ReadonlyMap<string, Readonly<{ scrollX: number; scrollY: number }>> =
+      this.scrollOverrides,
+  ): VNode {
```
[the remark this answers is not in the document]
Comment (RtlZeroMemory) [reply packages/core/src/app/widgetRenderer.ts:2474]: Ran Biome formatting against `packages/core/src/app/widgetRenderer.ts` in the current branch state; no formatter delta remained. This thread is addressed in the latest commit set.

**Comment by RtlZeroMemory** [comment]
Follow-up correction for clarity:

Addressed remaining nitpicks in commit `b10c497`:
- Added settle-cap observability metric/audit in `widgetRenderer` settle loop.
- Strengthened hidden-wrapper edge-case assertions with zero-rect checks.
- Added explicit rect assertions for interleaved absolute/flow order tests.
- Added focused `flexBasis` and `flexShrink` behavioral cases in flex-distribution tests.

Local validation after patch: lint ✅, build ✅, filtered suites (182/182) ✅.
